hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntityResultImpl.java

```diff
@@ -50,7 +52,8 @@ public NavigablePath resolveNavigablePath(Fetchable fetchable) {
 				final NavigablePath navigablePath = tableGroupJoin.getNavigablePath();
 				if ( tableGroupJoin.getJoinedGroup().isFetched()
 						&& fetchable.getFetchableName().equals( navigablePath.getLocalName() )
-						&& tableGroupJoin.getJoinedGroup().getModelPart() == fetchable ) {
+						&& tableGroupJoin.getJoinedGroup().getModelPart() == fetchable
+						&& castNonNull( navigablePath.getParent() ).equals( getNavigablePath() ) ) {
```

Comment:
Are `AggregateEmbeddableFetchImpl` and `EmbeddableFetchImpl` also affected by this? They use the same code.

Reply:
Probably, I've applied the same change to both code paths so that we're consistent when calling `resolveNavigablePath`. Thanks!